src/components/tools/FrequencyCalculator.tsx

```diff
@@ -326,12 +354,22 @@ export const FrequencyCalculator = (): JSX.Element => {
         selectedModemPreset.bw / 2000 +
         updatedChannel * (selectedModemPreset.bw / 1000),
     );
+
+    // Calculate the default slot
+    const channelName = formatChannelName(
+      Protobuf.Config.Config_LoRaConfig_ModemPreset[modemPreset],
+    ); // Example: LONG_FAST -> LongFast
+    const defaultSlot = determineFrequencySlot(
+      channelName,
+      calculatedNumChannels,
+    );
+    setDefaultSlot(defaultSlot);
   }, [modemPreset, region, channel]);
 
   return (
-    <div className="flex flex-col border-l-[5px] shadow-md my-4 border-accent rounded-lg p-4 bg-secondary gap-2">
-      <div className="flex gap-2">
-        <label htmlFor="modemPreset">Modem Preset:</label>
+    <div class="flex flex-col border-l-[5px] shadow-md my-4 border-accent rounded-lg p-4 bg-secondary gap-2">
+      <div class="flex gap-2">
+        <label for="modemPreset">Modem Preset:</label>
```

Comment:
These changes from `className` and `htmlFor` to the HTML equivalents won't work right in JSX, most likely. I could be wrong about that though.